src/TAlias.cpp

```diff
@@ -323,35 +349,49 @@ void TAlias::compileAll()
     mNeedsToBeCompiled = true;
     if( ! compileScript() )
     {
-        if( mudlet::debugMode ) {TDebug(QColor(Qt::white),QColor(Qt::red))<<"ERROR: Lua compile error. compiling script of alias:"<<mName<<"\n">>0;}
+        if( mudlet::debugMode ) {
+            TDebug( QColor(Qt::white), QColor(Qt::red) ) << "LUA ERROR: when compiling script of alias:"
+                                                         << mName
+                                                         << "\n"
+                                                         >> 0;
+        }
         mOK_code = false;
     }
-    typedef list<TAlias *>::const_iterator I;
-    for( I it = mpMyChildrenList->begin(); it != mpMyChildrenList->end(); it++)
-    {
-        TAlias * pChild = *it;
-        pChild->compileAll();
-    }
-}
-
-void TAlias::compile()
-{
-    if( mNeedsToBeCompiled )
+    else
     {
-        if( ! compileScript() )
-        {
-            if( mudlet::debugMode ) {TDebug(QColor(Qt::white),QColor(Qt::red))<<"ERROR: Lua compile error. compiling script of alias:"<<mName<<"\n">>0;}
-            mOK_code = false;
-        }
+        TDebug( QColor(Qt::white), QColor(Qt::red) ) << "LUA OK: when compiling script of alias:"
+                                                     << mName
+                                                     << "\n"
+                                                     >> 0;
     }
+    compileRegex(); // Effectively will repost the error if there was a problem in the regex
     typedef list<TAlias *>::const_iterator I;
     for( I it = mpMyChildrenList->begin(); it != mpMyChildrenList->end(); it++)
     {
         TAlias * pChild = *it;
-        pChild->compile();
+        pChild->compileAll();
     }
 }
 
+// Not used - so commented out rather than updating error/success messages
```

Comment:
But it isn't a __bug fix__, am I allowed to do it in the release_30 branch? :wink:

Also, I think there is similar duplication in all the items! The difference between `compile()` and `compileAll()` is that the latter compiles the code unconditionally (sets `mNeedsToBeCompiled` before running the code through the compiler which then resets it on success) whereas the former _only_ compiles the code if the flag is set -  they both call themselves recursively on the children of their item. :smirk:

It would actually make sense to merge their functionality as `compile( const bool isForced = false )`...